pkg/workspace/data_values_pre_processing.go

```diff
@@ -149,3 +158,26 @@ func (p DataValuesPreProcessing) overlayValuesOverlays(valuesDoc *yamlmeta.Docum
 
 	return result, nil
 }
+
+func isLibDataDoc(doc *yamlmeta.Document) (bool, error) {
```

Comment:
lets throw this function on DataValuesPreProcessing. if it doesnt need access to anything on it, then you can declare it like so: `func (DataValuesPreProcessing) isLibDataDoc...`

Reply:
we should also figure out how to consolidate logic in this function with GetValuesForLibraryAndChildren. they both are doing same parsing and arguably should be doing same validation. it could be with a new class eg DataValuesDoc{doc}, that has appropriate methods. we definitely want to ensure that validation of kwards and library value format is done as part of .apply() to throw invalid data/values declaration errors early on (not conditioned on calling to library module .eval())